sa/storage-authority.go

```diff
@@ -673,6 +673,125 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	return
 }
 
+const collectAuthIDs = `SELECT id FROM %s
+WHERE identifier = :ident AND
+status != :invalid AND
+status != :revoked AND
+expires > :now`
+
+// RevokeAuthorizationsByDomain invalidates all pending or finalized authorization
+// and any associated challenges for a single domain
+func (ssa *SQLStorageAuthority) RevokeAuthorizationsByDomain(ident core.AcmeIdentifier) (int64, int64, error) {
```

Comment:
Yeah, I meant putting a LIMIT in these queries to prevent us from consuming an entire MariaDB thread for possibly a very long time and then looping around until we zero them out.

We've already got folks with many thousands of pending authorizations, for instance. Those numbers will continue to climb.

Basically, we can't just do this in one big transaction. Which means we'll also have to put a little more taught into what happens when people make some pending authorizations or authorizations while we're revoking them. Possibly we need to gather all the authorizations we revoked and then only revoke related pending authorizations or similar.